diff --git a/dlls/winewayland.drv/window.c b/dlls/winewayland.drv/window.c
index ce5b6bb..e09cf74 100644
--- a/dlls/winewayland.drv/window.c
+++ b/dlls/winewayland.drv/window.c
@@ -63,9 +63,14 @@ static struct wayland_win_data *wayland_win_data_create(HWND hwnd, const struct
     struct rb_entry *rb_entry;
     HWND parent;
 
-    /* Don't create win data for desktop or HWND_MESSAGE windows. */
-    if (!(parent = NtUserGetAncestor(hwnd, GA_PARENT))) return NULL;
-    if (parent != NtUserGetDesktopWindow() && !NtUserGetAncestor(parent, GA_PARENT))
+    /* Don't create win data for HWND_MESSAGE windows.
+     * Allow the desktop window for virtual desktop support, so that
+     * it gets a Wayland surface and can receive pointer events. */
+    if (!(parent = NtUserGetAncestor(hwnd, GA_PARENT)))
+    {
+        if (hwnd != NtUserGetDesktopWindow()) return NULL;
+    }
+    else if (parent != NtUserGetDesktopWindow() && !NtUserGetAncestor(parent, GA_PARENT))
         return NULL;
 
     if (!(data = calloc(1, sizeof(*data)))) return NULL;
