diff --git a/dlls/win32u/winstation.c b/dlls/win32u/winstation.c
index eaa5f40..740734a 100644
--- a/dlls/win32u/winstation.c
+++ b/dlls/win32u/winstation.c
@@ -825,6 +825,10 @@ HWND get_desktop_window(void)
         PEB *peb = NtCurrentTeb()->Peb;
         HANDLE process, thread;
         unsigned int status;
+        HANDLE mutex = 0;
+        UNICODE_STRING mut_name;
+        OBJECT_ATTRIBUTES mut_attr;
+        WCHAR mut_buffer[MAX_PATH + 64];
 
         SERVER_START_REQ( set_user_object_info )
         {
@@ -842,47 +846,19 @@ HWND get_desktop_window(void)
         }
         SERVER_END_REQ;
 
-        params.Flags           = PROCESS_PARAMS_FLAG_NORMALIZED;
-        params.Environment     = peb->ProcessParameters->Environment;
-        params.EnvironmentSize = peb->ProcessParameters->EnvironmentSize;
-        params.hStdError       = peb->ProcessParameters->hStdError;
-        RtlInitUnicodeString( &params.CurrentDirectory.DosPath, system_dir );
-        RtlInitUnicodeString( &params.ImagePathName, appnameW + 4 );
-        RtlInitUnicodeString( &params.CommandLine, cmdlineW );
-        RtlInitUnicodeString( &params.WindowTitle, appnameW + 4 );
-        RtlInitUnicodeString( &params.Desktop, desktop );
-
-        ps_attr.TotalLength = sizeof(ps_attr);
-        ps_attr.Attributes[0].Attribute    = PS_ATTRIBUTE_IMAGE_NAME;
-        ps_attr.Attributes[0].Size         = sizeof(appnameW) - sizeof(WCHAR);
-        ps_attr.Attributes[0].ValuePtr     = (WCHAR *)appnameW;
-        ps_attr.Attributes[0].ReturnLength = NULL;
-
-        if (NtCurrentTeb64() && !NtCurrentTeb64()->TlsSlots[WOW64_TLS_FILESYSREDIR])
-        {
-            NtCurrentTeb64()->TlsSlots[WOW64_TLS_FILESYSREDIR] = TRUE;
-            status = NtCreateUserProcess( &process, &thread, PROCESS_ALL_ACCESS, THREAD_ALL_ACCESS,
-                                          NULL, NULL, 0, THREAD_CREATE_FLAGS_CREATE_SUSPENDED, &params,
-                                          &create_info, &ps_attr );
-            NtCurrentTeb64()->TlsSlots[WOW64_TLS_FILESYSREDIR] = FALSE;
-        }
-        else
-            status = NtCreateUserProcess( &process, &thread, PROCESS_ALL_ACCESS, THREAD_ALL_ACCESS,
-                                          NULL, NULL, 0, THREAD_CREATE_FLAGS_CREATE_SUSPENDED, &params,
-                                          &create_info, &ps_attr );
-        if (!status)
-        {
-            NtResumeThread( thread, NULL );
-            TRACE_(win)( "started explorer\n" );
-            NtUserWaitForInputIdle( process, 10000, FALSE );
-            NtClose( thread );
-            NtClose( process );
-        }
-        else ERR_(win)( "failed to start explorer %x\n", status );
+        static const WCHAR mut_prefixW[] = {'\\','B','a','s','e','N','a','m','e','d','O','b','j','e','c','t','s','\\','_','_','w','i','n','e','_','e','x','p','l','o','r','e','r','_','d','e','s','k','t','o','p','_','m','u','t','e','x','_',0};
+
+        wcscpy( mut_buffer, mut_prefixW );
+        wcscat( mut_buffer, desktop );
+        RtlInitUnicodeString( &mut_name, mut_buffer );
+        InitializeObjectAttributes( &mut_attr, &mut_name, OBJ_OPENIF, NULL, NULL );
+
+        if (!NtCreateMutant( &mutex, MUTANT_ALL_ACCESS, &mut_attr, FALSE ))
+            NtWaitForSingleObject( mutex, FALSE, NULL );
 
         SERVER_START_REQ( get_desktop_window )
         {
-            req->force = 1;
+            req->force = 0;
             if (!wine_server_call( req ))
             {
                 thread_info->top_window = reply->top_window;
@@ -890,6 +866,64 @@ HWND get_desktop_window(void)
             }
         }
         SERVER_END_REQ;
+
+        if (!thread_info->top_window)
+        {
+            params.Flags           = PROCESS_PARAMS_FLAG_NORMALIZED;
+            params.Environment     = peb->ProcessParameters->Environment;
+            params.EnvironmentSize = peb->ProcessParameters->EnvironmentSize;
+            params.hStdError       = peb->ProcessParameters->hStdError;
+            RtlInitUnicodeString( &params.CurrentDirectory.DosPath, system_dir );
+            RtlInitUnicodeString( &params.ImagePathName, appnameW + 4 );
+            RtlInitUnicodeString( &params.CommandLine, cmdlineW );
+            RtlInitUnicodeString( &params.WindowTitle, appnameW + 4 );
+            RtlInitUnicodeString( &params.Desktop, desktop );
+
+            ps_attr.TotalLength = sizeof(ps_attr);
+            ps_attr.Attributes[0].Attribute    = PS_ATTRIBUTE_IMAGE_NAME;
+            ps_attr.Attributes[0].Size         = sizeof(appnameW) - sizeof(WCHAR);
+            ps_attr.Attributes[0].ValuePtr     = (WCHAR *)appnameW;
+            ps_attr.Attributes[0].ReturnLength = NULL;
+
+            if (NtCurrentTeb64() && !NtCurrentTeb64()->TlsSlots[WOW64_TLS_FILESYSREDIR])
+            {
+                NtCurrentTeb64()->TlsSlots[WOW64_TLS_FILESYSREDIR] = TRUE;
+                status = NtCreateUserProcess( &process, &thread, PROCESS_ALL_ACCESS, THREAD_ALL_ACCESS,
+                                              NULL, NULL, 0, THREAD_CREATE_FLAGS_CREATE_SUSPENDED, &params,
+                                              &create_info, &ps_attr );
+                NtCurrentTeb64()->TlsSlots[WOW64_TLS_FILESYSREDIR] = FALSE;
+            }
+            else
+                status = NtCreateUserProcess( &process, &thread, PROCESS_ALL_ACCESS, THREAD_ALL_ACCESS,
+                                              NULL, NULL, 0, THREAD_CREATE_FLAGS_CREATE_SUSPENDED, &params,
+                                              &create_info, &ps_attr );
+            if (!status)
+            {
+                NtResumeThread( thread, NULL );
+                TRACE_(win)( "started explorer\n" );
+                NtUserWaitForInputIdle( process, 10000, FALSE );
+                NtClose( thread );
+                NtClose( process );
+            }
+            else ERR_(win)( "failed to start explorer %x\n", status );
+
+            SERVER_START_REQ( get_desktop_window )
+            {
+                req->force = 1;
+                if (!wine_server_call( req ))
+                {
+                    thread_info->top_window = reply->top_window;
+                    thread_info->msg_window = reply->msg_window;
+                }
+            }
+            SERVER_END_REQ;
+        }
+
+        if (mutex)
+        {
+            NtReleaseMutant( mutex, NULL );
+            NtClose( mutex );
+        }
     }
 
     if (!thread_info->top_window) ERR_(win)( "failed to create desktop window\n" );
